noddos: fails to build

Maintainer: @StevenHessing 
Environment: OpenWrt HEAD

Description:

`noddos` fails to build with modern `ipset` library, after at least http://git.netfilter.org/ipset/commit/?id=55fdd96e331e920ee62bd816a572ac24f6dcd1ae.  The compiler spew is as follows:

```
make[5]: Entering directory '/tank/openwrt/scratch/builder-usbnetgw2/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/noddos-0.5.5'
[  1%] Building CXX object src/CMakeFiles/Identifier_test.dir/DeviceProfile.cxx.o
In file included from /tank/openwrt/scratch/builder-usbnetgw2/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/noddos-0.5.5/src/DeviceProfile.h:44:0,
                 from /tank/openwrt/scratch/builder-usbnetgw2/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/noddos-0.5.5/src/DeviceProfile.cxx:23:
/tank/openwrt/scratch/builder-usbnetgw2/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/noddos-0.5.5/src/Ipset.h: In member function 'bool Ipset::Exists()':
/tank/openwrt/scratch/builder-usbnetgw2/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/noddos-0.5.5/src/Ipset.h:102:70: error: invalid conversion from'int (*)(const char*, ...)' to 'ipset_print_outfn {aka int (*)(ipset_session*, void*, const char*, ...)}' [-fpermissive]
             struct ipset_session *session = ipset_session_init(printf);
                                                                      ^
/tank/openwrt/scratch/builder-usbnetgw2/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/noddos-0.5.5/src/Ipset.h:102:70: error: too few arguments to function 'ipset_session* ipset_session_init(ipset_print_outfn, void*)'
In file included from /tank/openwrt/scratch/builder-usbnetgw2/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/noddos-0.5.5/src/Ipset.h:37:0,
                 from /tank/openwrt/scratch/builder-usbnetgw2/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/noddos-0.5.5/src/DeviceProfile.h:44,
                 from /tank/openwrt/scratch/builder-usbnetgw2/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/noddos-0.5.5/src/DeviceProfile.cxx:23:
/tank/openwrt/scratch/builder-usbnetgw2/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/usr/include/libipset/session.h:132:30: note: declared here
 extern struct ipset_session *ipset_session_init(ipset_print_outfn outfn,
                              ^~~~~~~~~~~~~~~~~~
In file included from /tank/openwrt/scratch/builder-usbnetgw2/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/noddos-0.5.5/src/DeviceProfile.h:44:0,
                 from /tank/openwrt/scratch/builder-usbnetgw2/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/noddos-0.5.5/src/DeviceProfile.cxx:23:
/tank/openwrt/scratch/builder-usbnetgw2/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/noddos-0.5.5/src/Ipset.h:109:17: error: 'ipset_envopt_parse' was not declared in this scope
             if (ipset_envopt_parse(session, IPSET_ENV_EXIST, NULL) < 0) {
                 ^~~~~~~~~~~~~~~~~~
/tank/openwrt/scratch/builder-usbnetgw2/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/noddos-0.5.5/src/Ipset.h:109:17: note: suggested alternative: 'ipset_envopt_unset'
             if (ipset_envopt_parse(session, IPSET_ENV_EXIST, NULL) < 0) {
                 ^~~~~~~~~~~~~~~~~~
                 ipset_envopt_unset
/tank/openwrt/scratch/builder-usbnetgw2/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/noddos-0.5.5/src/Ipset.h:116:95: error: 'ipset_session_error' was not declared in this scope
                 syslog (LOG_ERR, "Ipset: Can't commit for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
                                                                                               ^~~~~~~~~~~~~~~~~~~
/tank/openwrt/scratch/builder-usbnetgw2/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/noddos-0.5.5/src/Ipset.h:116:95: note: suggested alternative: 'ipset_session_report'
                 syslog (LOG_ERR, "Ipset: Can't commit for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
                                                                                               ^~~~~~~~~~~~~~~~~~~
                                                                                               ipset_session_report
src/CMakeFiles/Identifier_test.dir/build.make:75: recipe for target 'src/CMakeFiles/Identifier_test.dir/DeviceProfile.cxx.o' failed
make[5]: *** [src/CMakeFiles/Identifier_test.dir/DeviceProfile.cxx.o] Error 1
make[5]: Leaving directory '/tank/openwrt/scratch/builder-usbnetgw2/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/noddos-0.5.5'
```

> Can you describe what tests have you run, so that I can properly document the PR?

scource code:https://github.com/coolsnowwolf/lede.git
errors:

> make[3]: Leaving directory '/home/h/Desktop/lede/package/libs/libnetfilter-conntrack'
time: package/libs/libnetfilter-conntrack/compile#0.33#0.13#0.64
make[3]: Entering directory '/home/h/Desktop/lede/feeds/packages/net/noddos'
touch /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.prepared_8ee5294a600a7ff1d022b10fd10c30f6_6664517399ebbbc92a37c5bb081b5c53_check
. /home/h/Desktop/lede/include/shell.sh; xzcat /home/h/Desktop/lede/dl/noddos-0.5.5.tar.xz | tar -C /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.. -xf - 
[ ! -d ./src/ ] || cp -fpR ./src/. /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5
touch /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.prepared_8ee5294a600a7ff1d022b10fd10c30f6_6664517399ebbbc92a37c5bb081b5c53
rm -f /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.configured_*
rm -f /home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/stamp/.noddos_installed
make[3]: [Makefile:84: /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.configured_68b329da9893e34099c7d8ad5cb9c940] Error 123 (ignored)
mkdir -p /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5
(cd /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5; CFLAGS="-Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -iremap/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5:noddos-0.5.5 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro " CXXFLAGS="-Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -iremap/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5:noddos-0.5.5 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro " LDFLAGS="-L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/lib -znow -zrelro " cmake -DCMAKE_SYSTEM_NAME=Linux -DCMAKE_SYSTEM_VERSION=1 -DCMAKE_SYSTEM_PROCESSOR=mips -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_CXX_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_C_COMPILER="/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-gcc" -DCMAKE_C_COMPILER_ARG1="" -DCMAKE_CXX_COMPILER="/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-g++" -DCMAKE_CXX_COMPILER_ARG1="" -DCMAKE_ASM_COMPILER="/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-gcc" -DCMAKE_ASM_COMPILER_ARG1="" -DCMAKE_EXE_LINKER_FLAGS:STRING="-L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/lib -znow -zrelro" -DCMAKE_MODULE_LINKER_FLAGS:STRING="-L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/lib -znow -zrelro -Wl,-Bsymbolic-functions" -DCMAKE_SHARED_LINKER_FLAGS:STRING="-L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/lib -znow -zrelro -Wl,-Bsymbolic-functions" -DCMAKE_AR="/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-gcc-ar" -DCMAKE_NM="/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-gcc-nm" -DCMAKE_RANLIB="/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-gcc-ranlib" -DCMAKE_FIND_ROOT_PATH="/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr;/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl" -DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=BOTH -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY -DCMAKE_STRIP=: -DCMAKE_INSTALL_PREFIX=/usr -DDL_LIBRARY=/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl -DCMAKE_PREFIX_PATH=/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl -DCMAKE_SKIP_RPATH=TRUE  /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5 )
-- The C compiler identification is GNU 7.4.0
-- The CXX compiler identification is GNU 7.4.0
-- Check for working C compiler: /home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-gcc
-- Check for working C compiler: /home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-g++
-- Check for working CXX compiler: /home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_ASM_COMPILER
    CMAKE_ASM_COMPILER_ARG1
    CMAKE_FIND_ROOT_PATH_MODE_LIBRARY
    CMAKE_MODULE_LINKER_FLAGS
    CMAKE_SHARED_LINKER_FLAGS
    DL_LIBRARY


-- Build files have been written to: /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5
touch /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.configured_68b329da9893e34099c7d8ad5cb9c940
rm -f /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.built
touch /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.built_check
CFLAGS="-Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -iremap/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5:noddos-0.5.5 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro  -I/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/include -I/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/include -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/include -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/include/fortify -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/include " CXXFLAGS="-Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -iremap/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5:noddos-0.5.5 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro  -I/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/include -I/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/include -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/include -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/include/fortify -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/include " LDFLAGS="-L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/lib -znow -zrelro " make -j1 -C /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/. AR="mips-openwrt-linux-musl-gcc-ar" AS="mips-openwrt-linux-musl-gcc -c -Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -iremap/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5:noddos-0.5.5 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro" LD=mips-openwrt-linux-musl-ld NM="mips-openwrt-linux-musl-gcc-nm" CC="mips-openwrt-linux-musl-gcc" GCC="mips-openwrt-linux-musl-gcc" CXX="mips-openwrt-linux-musl-g++" RANLIB="mips-openwrt-linux-musl-gcc-ranlib" STRIP=mips-openwrt-linux-musl-strip OBJCOPY=mips-openwrt-linux-musl-objcopy OBJDUMP=mips-openwrt-linux-musl-objdump SIZE=mips-openwrt-linux-musl-size CROSS="mips-openwrt-linux-musl-" ARCH="mips" CMAKE_COMMAND='/home/h/Desktop/lede/staging_dir/host/bin/cmake' CMAKE_DISABLE_cmake_check_build_system=1 ;
make[4]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[5]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Scanning dependencies of target Identifier_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[  1%] Building CXX object src/CMakeFiles/Identifier_test.dir/Identifier_test.cxx.o
[  2%] Building CXX object src/CMakeFiles/Identifier_test.dir/DeviceProfile.cxx.o
In file included from /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/DeviceProfile.h:44:0,
                 from /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/DeviceProfile.cxx:23:
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/Ipset.h: In member function 'bool Ipset::Exists()':
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/Ipset.h:102:70: error: invalid conversion from 'int (*)(const char*, ...)' to 'ipset_print_outfn {aka int (*)(ipset_session*, void*, const char*, ...)}' [-fpermissive]
             struct ipset_session *session = ipset_session_init(printf);
                                                                      ^
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/Ipset.h:102:70: error: too few arguments to function 'ipset_session* ipset_session_init(ipset_print_outfn, void*)'
In file included from /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/Ipset.h:37:0,
                 from /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/DeviceProfile.h:44,
                 from /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/DeviceProfile.cxx:23:
/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/include/libipset/session.h:132:30: note: declared here
 extern struct ipset_session *ipset_session_init(ipset_print_outfn outfn,
                              ^~~~~~~~~~~~~~~~~~
In file included from /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/DeviceProfile.h:44:0,
                 from /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/DeviceProfile.cxx:23:
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/Ipset.h:109:17: error: 'ipset_envopt_parse' was not declared in this scope
             if (ipset_envopt_parse(session, IPSET_ENV_EXIST, NULL) < 0) {
                 ^~~~~~~~~~~~~~~~~~
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/Ipset.h:109:17: note: suggested alternative: 'ipset_envopt_unset'
             if (ipset_envopt_parse(session, IPSET_ENV_EXIST, NULL) < 0) {
                 ^~~~~~~~~~~~~~~~~~
                 ipset_envopt_unset
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/Ipset.h:116:95: error: 'ipset_session_error' was not declared in this scope
                 syslog (LOG_ERR, "Ipset: Can't commit for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
                                                                                               ^~~~~~~~~~~~~~~~~~~
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/Ipset.h:116:95: note: suggested alternative: 'ipset_session_report'
                 syslog (LOG_ERR, "Ipset: Can't commit for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
                                                                                               ^~~~~~~~~~~~~~~~~~~
                                                                                               ipset_session_report
make[6]: *** [src/CMakeFiles/Identifier_test.dir/build.make:76: src/CMakeFiles/Identifier_test.dir/DeviceProfile.cxx.o] Error 1
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[5]: *** [CMakeFiles/Makefile2:119: src/CMakeFiles/Identifier_test.dir/all] Error 2
make[5]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[4]: *** [Makefile:130: all] Error 2
make[4]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[3]: *** [Makefile:85: /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.built] Error 2
make[3]: Leaving directory '/home/h/Desktop/lede/feeds/packages/net/noddos'
time: package/feeds/packages/noddos/compile#4.07#0.53#12.82
make[2]: *** [package/Makefile:109: package/feeds/packages/noddos/compile] Error 2
make[2]: Leaving directory '/home/h/Desktop/lede'
make[1]: *** [package/Makefile:103: /home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/home/h/Desktop/lede'
make: *** [/home/h/Desktop/lede/include/toplevel.mk:218：world] 错误 2
h@h-PC:~/Desktop/lede$

I take place for files . 
sucessed, 
log :

> make[3]: Leaving directory '/home/h/Desktop/lede/package/libs/libnetfilter-conntrack'
time: package/libs/libnetfilter-conntrack/compile#0.36#0.12#0.56
make[3]: Entering directory '/home/h/Desktop/lede/feeds/packages/net/noddos'
touch /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.prepared_5d76d9e07f5f6d8b769c49f19512d24d_6664517399ebbbc92a37c5bb081b5c53_check
. /home/h/Desktop/lede/include/shell.sh; xzcat /home/h/Desktop/lede/dl/noddos-0.5.5.tar.xz | tar -C /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.. -xf - 
[ ! -d ./src/ ] || cp -fpR ./src/. /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5

Applying ./patches/010-openssl-1.1-fixes.patch using plaintext: 
patching file src/opensslfingerprint.cxx

Applying ./patches/020-Ipset.cxx-update-libipset-API-to-version-7.patch using plaintext: 
patching file src/Ipset.cxx
patching file src/Ipset.h
patch unexpectedly ends in middle of line
Hunk #2 succeeded at 124 with fuzz 1.
touch /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.prepared_5d76d9e07f5f6d8b769c49f19512d24d_6664517399ebbbc92a37c5bb081b5c53
rm -f /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.configured_*
rm -f /home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/stamp/.noddos_installed
make[3]: [Makefile:84: /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.configured_68b329da9893e34099c7d8ad5cb9c940] Error 123 (ignored)
mkdir -p /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5
(cd /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5; CFLAGS="-Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -iremap/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5:noddos-0.5.5 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro " CXXFLAGS="-Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -iremap/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5:noddos-0.5.5 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro " LDFLAGS="-L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/lib -znow -zrelro " cmake -DCMAKE_SYSTEM_NAME=Linux -DCMAKE_SYSTEM_VERSION=1 -DCMAKE_SYSTEM_PROCESSOR=mips -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_CXX_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_C_COMPILER="/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-gcc" -DCMAKE_C_COMPILER_ARG1="" -DCMAKE_CXX_COMPILER="/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-g++" -DCMAKE_CXX_COMPILER_ARG1="" -DCMAKE_ASM_COMPILER="/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-gcc" -DCMAKE_ASM_COMPILER_ARG1="" -DCMAKE_EXE_LINKER_FLAGS:STRING="-L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/lib -znow -zrelro" -DCMAKE_MODULE_LINKER_FLAGS:STRING="-L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/lib -znow -zrelro -Wl,-Bsymbolic-functions" -DCMAKE_SHARED_LINKER_FLAGS:STRING="-L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/lib -znow -zrelro -Wl,-Bsymbolic-functions" -DCMAKE_AR="/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-gcc-ar" -DCMAKE_NM="/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-gcc-nm" -DCMAKE_RANLIB="/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-gcc-ranlib" -DCMAKE_FIND_ROOT_PATH="/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr;/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl" -DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=BOTH -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY -DCMAKE_STRIP=: -DCMAKE_INSTALL_PREFIX=/usr -DDL_LIBRARY=/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl -DCMAKE_PREFIX_PATH=/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl -DCMAKE_SKIP_RPATH=TRUE  /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5 )
-- The C compiler identification is GNU 7.4.0
-- The CXX compiler identification is GNU 7.4.0
-- Check for working C compiler: /home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-gcc
-- Check for working C compiler: /home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-g++
-- Check for working CXX compiler: /home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/bin/mips-openwrt-linux-musl-g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_ASM_COMPILER
    CMAKE_ASM_COMPILER_ARG1
    CMAKE_FIND_ROOT_PATH_MODE_LIBRARY
    CMAKE_MODULE_LINKER_FLAGS
    CMAKE_SHARED_LINKER_FLAGS
    DL_LIBRARY


-- Build files have been written to: /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5
touch /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.configured_68b329da9893e34099c7d8ad5cb9c940
rm -f /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.built
touch /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.built_check
CFLAGS="-Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -iremap/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5:noddos-0.5.5 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro  -I/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/include -I/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/include -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/include -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/include/fortify -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/include " CXXFLAGS="-Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -iremap/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5:noddos-0.5.5 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro  -I/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/include -I/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/include -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/include -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/include/fortify -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/include " LDFLAGS="-L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/lib -znow -zrelro " make -j1 -C /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/. AR="mips-openwrt-linux-musl-gcc-ar" AS="mips-openwrt-linux-musl-gcc -c -Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -iremap/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5:noddos-0.5.5 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro" LD=mips-openwrt-linux-musl-ld NM="mips-openwrt-linux-musl-gcc-nm" CC="mips-openwrt-linux-musl-gcc" GCC="mips-openwrt-linux-musl-gcc" CXX="mips-openwrt-linux-musl-g++" RANLIB="mips-openwrt-linux-musl-gcc-ranlib" STRIP=mips-openwrt-linux-musl-strip OBJCOPY=mips-openwrt-linux-musl-objcopy OBJDUMP=mips-openwrt-linux-musl-objdump SIZE=mips-openwrt-linux-musl-size CROSS="mips-openwrt-linux-musl-" ARCH="mips" CMAKE_COMMAND='/home/h/Desktop/lede/staging_dir/host/bin/cmake' CMAKE_DISABLE_cmake_check_build_system=1 ;
make[4]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[5]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Scanning dependencies of target Identifier_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[  1%] Building CXX object src/CMakeFiles/Identifier_test.dir/Identifier_test.cxx.o
[  2%] Building CXX object src/CMakeFiles/Identifier_test.dir/DeviceProfile.cxx.o
[  3%] Building CXX object src/CMakeFiles/Identifier_test.dir/Ipset.cxx.o
[  4%] Linking CXX executable Identifier_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[  4%] Built target Identifier_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Scanning dependencies of target SsdpServer_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[  5%] Building CXX object src/CMakeFiles/SsdpServer_test.dir/SsdpServer_test.cxx.o
[  6%] Building CXX object src/CMakeFiles/SsdpServer_test.dir/HostCache.cxx.o
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/HostCache.cxx: In member function 'void HostCache::addorupdateDnsQueryCache(uint16_t)':
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/HostCache.cxx:380:89: warning: format '%zu' expects argument of type 'size_t', but argument 4 has type 'time_t {aka long int}' [-Wformat=]
      syslog (LOG_DEBUG, "HostCache: setting DnsQueryCache for %u to %zu", id, Expiration);
                                                                                         ^
[  7%] Building CXX object src/CMakeFiles/SsdpServer_test.dir/Host.cxx.o
[  8%] Building CXX object src/CMakeFiles/SsdpServer_test.dir/SsdpServer.cxx.o
[  9%] Building CXX object src/CMakeFiles/SsdpServer_test.dir/SsdpLocation.cxx.o
[ 10%] Building CXX object src/CMakeFiles/SsdpServer_test.dir/opensslfingerprint.cxx.o
[ 11%] Building CXX object src/CMakeFiles/SsdpServer_test.dir/Ipset.cxx.o
[ 12%] Building CXX object src/CMakeFiles/SsdpServer_test.dir/DeviceProfile.cxx.o
[ 13%] Linking CXX executable SsdpServer_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 13%] Built target SsdpServer_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Scanning dependencies of target Curl_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 14%] Building CXX object src/CMakeFiles/Curl_test.dir/Curl_test.cxx.o
[ 15%] Linking CXX executable Curl_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 15%] Built target Curl_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Scanning dependencies of target Config_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 16%] Building CXX object src/CMakeFiles/Config_test.dir/Config_test.cxx.o
[ 17%] Building CXX object src/CMakeFiles/Config_test.dir/opensslfingerprint.cxx.o
[ 18%] Linking CXX executable Config_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 18%] Built target Config_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Scanning dependencies of target noddos_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 19%] Building CXX object src/CMakeFiles/noddos_test.dir/noddos_test.cxx.o
[ 20%] Linking CXX executable noddos_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 20%] Built target noddos_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Scanning dependencies of target HostCache_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 21%] Building CXX object src/CMakeFiles/HostCache_test.dir/HostCache_test.cxx.o
[ 22%] Building CXX object src/CMakeFiles/HostCache_test.dir/Ipset.cxx.o
[ 23%] Building CXX object src/CMakeFiles/HostCache_test.dir/Host.cxx.o
[ 24%] Building CXX object src/CMakeFiles/HostCache_test.dir/HostCache.cxx.o
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/HostCache.cxx: In member function 'void HostCache::addorupdateDnsQueryCache(uint16_t)':
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/HostCache.cxx:380:89: warning: format '%zu' expects argument of type 'size_t', but argument 4 has type 'time_t {aka long int}' [-Wformat=]
      syslog (LOG_DEBUG, "HostCache: setting DnsQueryCache for %u to %zu", id, Expiration);
                                                                                         ^
[ 25%] Building CXX object src/CMakeFiles/HostCache_test.dir/SsdpServer.cxx.o
[ 26%] Building CXX object src/CMakeFiles/HostCache_test.dir/SsdpLocation.cxx.o
[ 27%] Building CXX object src/CMakeFiles/HostCache_test.dir/DeviceProfile.cxx.o
[ 28%] Building CXX object src/CMakeFiles/HostCache_test.dir/opensslfingerprint.cxx.o
[ 29%] Linking CXX executable HostCache_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 29%] Built target HostCache_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Scanning dependencies of target WsDiscovery_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 30%] Building CXX object src/CMakeFiles/WsDiscovery_test.dir/WsDiscovery_test.cxx.o
[ 31%] Building CXX object src/CMakeFiles/WsDiscovery_test.dir/HostCache.cxx.o
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/HostCache.cxx: In member function 'void HostCache::addorupdateDnsQueryCache(uint16_t)':
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/HostCache.cxx:380:89: warning: format '%zu' expects argument of type 'size_t', but argument 4 has type 'time_t {aka long int}' [-Wformat=]
      syslog (LOG_DEBUG, "HostCache: setting DnsQueryCache for %u to %zu", id, Expiration);
                                                                                         ^
[ 32%] Building CXX object src/CMakeFiles/WsDiscovery_test.dir/Host.cxx.o
[ 33%] Building CXX object src/CMakeFiles/WsDiscovery_test.dir/WsDiscovery.cxx.o
[ 34%] Building CXX object src/CMakeFiles/WsDiscovery_test.dir/opensslfingerprint.cxx.o
[ 35%] Building CXX object src/CMakeFiles/WsDiscovery_test.dir/Ipset.cxx.o
[ 36%] Building CXX object src/CMakeFiles/WsDiscovery_test.dir/DeviceProfile.cxx.o
[ 37%] Linking CXX executable WsDiscovery_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 37%] Built target WsDiscovery_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Scanning dependencies of target noddos
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 38%] Building CXX object src/CMakeFiles/noddos.dir/noddos.cxx.o
[ 39%] Building CXX object src/CMakeFiles/noddos.dir/Host.cxx.o
[ 40%] Building CXX object src/CMakeFiles/noddos.dir/Mdns.cxx.o
[ 41%] Building CXX object src/CMakeFiles/noddos.dir/HostCache.cxx.o
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/HostCache.cxx: In member function 'void HostCache::addorupdateDnsQueryCache(uint16_t)':
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/HostCache.cxx:380:89: warning: format '%zu' expects argument of type 'size_t', but argument 4 has type 'time_t {aka long int}' [-Wformat=]
      syslog (LOG_DEBUG, "HostCache: setting DnsQueryCache for %u to %zu", id, Expiration);
                                                                                         ^
[ 42%] Building CXX object src/CMakeFiles/noddos.dir/SsdpServer.cxx.o
[ 43%] Building CXX object src/CMakeFiles/noddos.dir/SsdpLocation.cxx.o
[ 44%] Building CXX object src/CMakeFiles/noddos.dir/FlowTrack.cxx.o
[ 45%] Building CXX object src/CMakeFiles/noddos.dir/Config.cxx.o
[ 46%] Building CXX object src/CMakeFiles/noddos.dir/opensslfingerprint.cxx.o
[ 47%] Building CXX object src/CMakeFiles/noddos.dir/PacketSnoop.cxx.o
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/PacketSnoop.cxx: In member function 'bool PacketSnoop::Parse(unsigned char*)':
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/PacketSnoop.cxx:398:87: warning: format '%p' expects argument of type 'void*', but argument 3 has type 'std::shared_ptr<TcpSnoop>' [-Wformat=]
                     syslog (LOG_DEBUG, "Creating TcpSnoop shared pointer at %p", tsPtr);
                                                                                       ^
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/PacketSnoop.cxx: In member function 'void PacketSnoop::addTcpSnoopInstance(Tins::IPv4Address, uint16_t, Tins::IPv4Address, uint16_t, std::shared_ptr<TcpSnoop>)':
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/PacketSnoop.cxx:513:74: warning: format '%p' expects argument of type 'void*', but argument 3 has type 'const std::shared_ptr<TcpSnoop>' [-Wformat=]
         syslog (LOG_DEBUG, "Adding TcpSnoop shared pointer at %p", ts_ptr);
                                                                          ^
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/PacketSnoop.cxx: In member function 'void PacketSnoop::addTcpSnoopInstance(Tins::IPv6Address, uint16_t, Tins::IPv6Address, uint16_t, std::shared_ptr<TcpSnoop>)':
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/PacketSnoop.cxx:522:74: warning: format '%p' expects argument of type 'void*', but argument 3 has type 'const std::shared_ptr<TcpSnoop>' [-Wformat=]
         syslog (LOG_DEBUG, "Adding TcpSnoop shared pointer at %p", ts_ptr);
                                                                          ^
[ 48%] Building CXX object src/CMakeFiles/noddos.dir/Ipset.cxx.o
[ 50%] Building CXX object src/CMakeFiles/noddos.dir/DeviceProfile.cxx.o
[ 51%] Building CXX object src/CMakeFiles/noddos.dir/WsDiscovery.cxx.o
[ 52%] Linking CXX executable noddos
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 52%] Built target noddos
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Scanning dependencies of target DeviceProfile_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 53%] Building CXX object src/CMakeFiles/DeviceProfile_test.dir/DeviceProfile_test.cxx.o
[ 54%] Building CXX object src/CMakeFiles/DeviceProfile_test.dir/DeviceProfile.cxx.o
[ 55%] Building CXX object src/CMakeFiles/DeviceProfile_test.dir/FlowTrack.cxx.o
[ 56%] Building CXX object src/CMakeFiles/DeviceProfile_test.dir/Config.cxx.o
[ 57%] Building CXX object src/CMakeFiles/DeviceProfile_test.dir/HostCache.cxx.o
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/HostCache.cxx: In member function 'void HostCache::addorupdateDnsQueryCache(uint16_t)':
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/HostCache.cxx:380:89: warning: format '%zu' expects argument of type 'size_t', but argument 4 has type 'time_t {aka long int}' [-Wformat=]
      syslog (LOG_DEBUG, "HostCache: setting DnsQueryCache for %u to %zu", id, Expiration);
                                                                                         ^
[ 58%] Building CXX object src/CMakeFiles/DeviceProfile_test.dir/Host.cxx.o
[ 59%] Building CXX object src/CMakeFiles/DeviceProfile_test.dir/SsdpServer.cxx.o
[ 60%] Building CXX object src/CMakeFiles/DeviceProfile_test.dir/SsdpLocation.cxx.o
[ 61%] Building CXX object src/CMakeFiles/DeviceProfile_test.dir/opensslfingerprint.cxx.o
[ 62%] Building CXX object src/CMakeFiles/DeviceProfile_test.dir/Ipset.cxx.o
[ 63%] Linking CXX executable DeviceProfile_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 63%] Built target DeviceProfile_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Scanning dependencies of target Host_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 64%] Building CXX object src/CMakeFiles/Host_test.dir/Host_test.cxx.o
[ 65%] Building CXX object src/CMakeFiles/Host_test.dir/Host.cxx.o
[ 66%] Building CXX object src/CMakeFiles/Host_test.dir/HostCache.cxx.o
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/HostCache.cxx: In member function 'void HostCache::addorupdateDnsQueryCache(uint16_t)':
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/HostCache.cxx:380:89: warning: format '%zu' expects argument of type 'size_t', but argument 4 has type 'time_t {aka long int}' [-Wformat=]
      syslog (LOG_DEBUG, "HostCache: setting DnsQueryCache for %u to %zu", id, Expiration);
                                                                                         ^
[ 67%] Building CXX object src/CMakeFiles/Host_test.dir/SsdpServer.cxx.o
[ 68%] Building CXX object src/CMakeFiles/Host_test.dir/SsdpLocation.cxx.o
[ 69%] Building CXX object src/CMakeFiles/Host_test.dir/DeviceProfile.cxx.o
[ 70%] Building CXX object src/CMakeFiles/Host_test.dir/opensslfingerprint.cxx.o
[ 71%] Building CXX object src/CMakeFiles/Host_test.dir/Ipset.cxx.o
[ 72%] Linking CXX executable Host_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 72%] Built target Host_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Scanning dependencies of target Ipset_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 73%] Building CXX object src/CMakeFiles/Ipset_test.dir/Ipset_test.cxx.o
[ 74%] Building CXX object src/CMakeFiles/Ipset_test.dir/Ipset.cxx.o
[ 75%] Linking CXX executable Ipset_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 75%] Built target Ipset_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Scanning dependencies of target Mdns_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 76%] Building CXX object src/CMakeFiles/Mdns_test.dir/Mdns_test.cxx.o
[ 77%] Building CXX object src/CMakeFiles/Mdns_test.dir/Mdns.cxx.o
[ 78%] Building CXX object src/CMakeFiles/Mdns_test.dir/HostCache.cxx.o
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/HostCache.cxx: In member function 'void HostCache::addorupdateDnsQueryCache(uint16_t)':
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/HostCache.cxx:380:89: warning: format '%zu' expects argument of type 'size_t', but argument 4 has type 'time_t {aka long int}' [-Wformat=]
      syslog (LOG_DEBUG, "HostCache: setting DnsQueryCache for %u to %zu", id, Expiration);
                                                                                         ^
[ 79%] Building CXX object src/CMakeFiles/Mdns_test.dir/Host.cxx.o
[ 80%] Building CXX object src/CMakeFiles/Mdns_test.dir/opensslfingerprint.cxx.o
[ 81%] Building CXX object src/CMakeFiles/Mdns_test.dir/Ipset.cxx.o
[ 82%] Building CXX object src/CMakeFiles/Mdns_test.dir/DeviceProfile.cxx.o
[ 83%] Linking CXX executable Mdns_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 83%] Built target Mdns_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Scanning dependencies of target MacAddress_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 84%] Building CXX object src/CMakeFiles/MacAddress_test.dir/MacAddress_test.cxx.o
[ 85%] Linking CXX executable MacAddress_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 85%] Built target MacAddress_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Scanning dependencies of target DnsCache_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 86%] Building CXX object src/CMakeFiles/DnsCache_test.dir/DnsCache_test.cxx.o
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/DnsCache_test.cxx: In function 'int main()':
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/DnsCache_test.cxx:74:100: warning: format '%lu' expects argument of type 'long unsigned int', but argument 3 has type 'std::vector<std::__cxx11::basic_string<char> >::size_type {aka unsigned int}' [-Wformat=]
         syslog (LOG_DEBUG,"Simple reverse lookup for 23.41.176.89 returned %lu fqdns", fqdns.size());
                                                                                        ~~~~~~~~~~~~^
[ 87%] Building CXX object src/CMakeFiles/DnsCache_test.dir/DeviceProfile.cxx.o
[ 88%] Building CXX object src/CMakeFiles/DnsCache_test.dir/Ipset.cxx.o
[ 89%] Linking CXX executable DnsCache_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 89%] Built target DnsCache_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Scanning dependencies of target PacketSnoop_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 90%] Building CXX object src/CMakeFiles/PacketSnoop_test.dir/PacketSnoop_test.cxx.o
[ 91%] Building CXX object src/CMakeFiles/PacketSnoop_test.dir/PacketSnoop.cxx.o
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/PacketSnoop.cxx: In member function 'bool PacketSnoop::Parse(unsigned char*)':
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/PacketSnoop.cxx:398:87: warning: format '%p' expects argument of type 'void*', but argument 3 has type 'std::shared_ptr<TcpSnoop>' [-Wformat=]
                     syslog (LOG_DEBUG, "Creating TcpSnoop shared pointer at %p", tsPtr);
                                                                                       ^
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/PacketSnoop.cxx: In member function 'void PacketSnoop::addTcpSnoopInstance(Tins::IPv4Address, uint16_t, Tins::IPv4Address, uint16_t, std::shared_ptr<TcpSnoop>)':
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/PacketSnoop.cxx:513:74: warning: format '%p' expects argument of type 'void*', but argument 3 has type 'const std::shared_ptr<TcpSnoop>' [-Wformat=]
         syslog (LOG_DEBUG, "Adding TcpSnoop shared pointer at %p", ts_ptr);
                                                                          ^
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/PacketSnoop.cxx: In member function 'void PacketSnoop::addTcpSnoopInstance(Tins::IPv6Address, uint16_t, Tins::IPv6Address, uint16_t, std::shared_ptr<TcpSnoop>)':
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/PacketSnoop.cxx:522:74: warning: format '%p' expects argument of type 'void*', but argument 3 has type 'const std::shared_ptr<TcpSnoop>' [-Wformat=]
         syslog (LOG_DEBUG, "Adding TcpSnoop shared pointer at %p", ts_ptr);
                                                                          ^
[ 92%] Building CXX object src/CMakeFiles/PacketSnoop_test.dir/HostCache.cxx.o
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/HostCache.cxx: In member function 'void HostCache::addorupdateDnsQueryCache(uint16_t)':
/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/HostCache.cxx:380:89: warning: format '%zu' expects argument of type 'size_t', but argument 4 has type 'time_t {aka long int}' [-Wformat=]
      syslog (LOG_DEBUG, "HostCache: setting DnsQueryCache for %u to %zu", id, Expiration);
                                                                                         ^
[ 93%] Building CXX object src/CMakeFiles/PacketSnoop_test.dir/Host.cxx.o
[ 94%] Building CXX object src/CMakeFiles/PacketSnoop_test.dir/SsdpServer.cxx.o
[ 95%] Building CXX object src/CMakeFiles/PacketSnoop_test.dir/SsdpLocation.cxx.o
[ 96%] Building CXX object src/CMakeFiles/PacketSnoop_test.dir/opensslfingerprint.cxx.o
[ 97%] Building CXX object src/CMakeFiles/PacketSnoop_test.dir/DeviceProfile.cxx.o
[ 98%] Building CXX object src/CMakeFiles/PacketSnoop_test.dir/Ipset.cxx.o
[100%] Linking CXX executable PacketSnoop_test
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[100%] Built target PacketSnoop_test
make[5]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[4]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
CFLAGS="-Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -iremap/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5:noddos-0.5.5 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro  -I/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/include -I/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/include -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/include -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/include/fortify -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/include " CXXFLAGS="-Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -iremap/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5:noddos-0.5.5 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro  -I/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/include -I/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/include -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/include -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/include/fortify -I/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/include " LDFLAGS="-L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/usr/lib -L/home/h/Desktop/lede/staging_dir/toolchain-mips_24kc_gcc-7.4.0_musl/lib -znow -zrelro " make -C /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/. AR="mips-openwrt-linux-musl-gcc-ar" AS="mips-openwrt-linux-musl-gcc -c -Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -iremap/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5:noddos-0.5.5 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro" LD=mips-openwrt-linux-musl-ld NM="mips-openwrt-linux-musl-gcc-nm" CC="mips-openwrt-linux-musl-gcc" GCC="mips-openwrt-linux-musl-gcc" CXX="mips-openwrt-linux-musl-g++" RANLIB="mips-openwrt-linux-musl-gcc-ranlib" STRIP=mips-openwrt-linux-musl-strip OBJCOPY=mips-openwrt-linux-musl-objcopy OBJDUMP=mips-openwrt-linux-musl-objdump SIZE=mips-openwrt-linux-musl-size CROSS="mips-openwrt-linux-musl-" ARCH="mips" CMAKE_COMMAND='/home/h/Desktop/lede/staging_dir/host/bin/cmake' CMAKE_DISABLE_cmake_check_build_system=1 DESTDIR="/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-install"  install;
make[4]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[5]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[  4%] Built target Identifier_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 13%] Built target SsdpServer_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 15%] Built target Curl_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 18%] Built target Config_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 20%] Built target noddos_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 29%] Built target HostCache_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 37%] Built target WsDiscovery_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 52%] Built target noddos
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 63%] Built target DeviceProfile_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 72%] Built target Host_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 75%] Built target Ipset_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 83%] Built target Mdns_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 85%] Built target MacAddress_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[ 89%] Built target DnsCache_test
make[6]: Entering directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
make[6]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
[100%] Built target PacketSnoop_test
make[5]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
Install the project...
-- Install configuration: "Release"
-- Installing: /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-install/usr/usr/sbin/noddos
make[4]: Leaving directory '/home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5'
touch /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.built
rm -rf /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos.installed /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos
mkdir -p /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos
install -d -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos/usr/bin
install -d -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos/usr/sbin
install -d -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos/etc/config
install -d -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos/etc/init.d
install -d -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos/etc/noddos
install -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/noddos /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos/usr/sbin/
install -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/tools/getnoddosdeviceprofiles /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos/usr/bin
install -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/tools/makenoddoscert.sh /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos/usr/bin
install -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/files/noddos.init /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos/etc/init.d/noddos
install -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/files/noddos.uciconfig /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos/etc/config/noddos
install -m0644 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/files/noddos.yml-base /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos/etc/noddos
install -m0644 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/files/noddosconfig.pem /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos/etc/noddos
touch /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos.installed
mkdir -p /home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/root-ar71xx/stamp
SHELL= flock /home/h/Desktop/lede/tmp/.root-copy.flock -c 'cp -fpR /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/.pkgdir/noddos/. /home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/root-ar71xx/'
touch /home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/root-ar71xx/stamp/.noddos_installed
mkdir -p /home/h/Desktop/lede/bin/targets/ar71xx/tiny/packages /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos/CONTROL /home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/pkginfo
install -d -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos/usr/bin
install -d -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos/usr/sbin
install -d -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos/etc/config
install -d -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos/etc/init.d
install -d -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos/etc/noddos
install -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/src/noddos /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos/usr/sbin/
install -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/tools/getnoddosdeviceprofiles /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos/usr/bin
install -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/tools/makenoddoscert.sh /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos/usr/bin
install -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/files/noddos.init /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos/etc/init.d/noddos
install -m0755 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/files/noddos.uciconfig /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos/etc/config/noddos
install -m0644 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/files/noddos.yml-base /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos/etc/noddos
install -m0644 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/files/noddosconfig.pem /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos/etc/noddos
find /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos -name 'CVS' -o -name '.svn' -o -name '.#*' -o -name '*~'| xargs -r rm -rf
export CROSS="mips-openwrt-linux-musl-"   ; NM="mips-openwrt-linux-musl-nm" STRIP="/home/h/Desktop/lede/staging_dir/host/bin/sstrip" STRIP_KMOD="/home/h/Desktop/lede/scripts/strip-kmod.sh" PATCHELF="/home/h/Desktop/lede/staging_dir/host/bin/patchelf" /home/h/Desktop/lede/scripts/rstrip.sh /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos
rstrip.sh: /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos/usr/sbin/noddos: executable
(cd /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos/CONTROL; ( echo "$CONTROL"; printf "Description: "; echo "$DESCRIPTION" | sed -e 's,^[[:space:]]*, ,g'; ) > control; chmod 644 control; ( echo "#!/bin/sh"; echo "[ \"\${IPKG_NO_SCRIPT}\" = \"1\" ] && exit 0"; echo "[ -x "\${IPKG_INSTROOT}/lib/functions.sh" ] || exit 0"; echo ". \${IPKG_INSTROOT}/lib/functions.sh"; echo "default_postinst \$0 \$@"; ) > postinst; ( echo "#!/bin/sh"; echo "[ -x "\${IPKG_INSTROOT}/lib/functions.sh" ] || exit 0"; echo ". \${IPKG_INSTROOT}/lib/functions.sh"; echo "default_prerm \$0 \$@"; ) > prerm; chmod 0755 postinst prerm; echo "$V_Package_noddos_conffiles" > conffiles;  echo "$V_Package_noddos_prerm" > prerm-pkg; chmod 0755 prerm-pkg; echo "$V_Package_noddos_postrm" > postrm; chmod 0755 postrm; )
install -d -m0755 /home/h/Desktop/lede/bin/packages/mips_24kc/packages
/home/h/Desktop/lede/scripts/ipkg-build -c -o 0 -g 0 /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos /home/h/Desktop/lede/bin/packages/mips_24kc/packages
/home/h/Desktop/lede/staging_dir/host/bin/find: '/etc/config/noddos': No such file or directory
Packaged contents of /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-mips_24kc/noddos into /home/h/Desktop/lede/bin/packages/mips_24kc/packages/noddos_0.5.5-2_mips_24kc.ipk
rm -rf /home/h/Desktop/lede/tmp/stage-noddos
mkdir -p /home/h/Desktop/lede/tmp/stage-noddos/host /home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/packages /home/h/Desktop/lede/staging_dir/host/packages
install -d -m0755 /home/h/Desktop/lede/tmp/stage-noddos
cp -fpR /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/noddos-0.5.5/ipkg-install/* /home/h/Desktop/lede/tmp/stage-noddos/
find /home/h/Desktop/lede/tmp/stage-noddos -name '*.la' | xargs -r rm -f; 
if [ -f /home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/packages/noddos.list ]; then /home/h/Desktop/lede/scripts/clean-package.sh "/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/packages/noddos.list" "/home/h/Desktop/lede/staging_dir/target-mips_24kc_musl"; fi
if [ -d /home/h/Desktop/lede/tmp/stage-noddos ]; then (cd /home/h/Desktop/lede/tmp/stage-noddos; find ./ > /home/h/Desktop/lede/tmp/stage-noddos.files); 	SHELL= flock /home/h/Desktop/lede/tmp/.staging-dir.flock -c ' mv /home/h/Desktop/lede/tmp/stage-noddos.files /home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/packages/noddos.list && cp -fpR /home/h/Desktop/lede/tmp/stage-noddos/* /home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/; '; fi
rm -rf /home/h/Desktop/lede/tmp/stage-noddos
touch /home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/stamp/.noddos_installed
make[3]: Leaving directory '/home/h/Desktop/lede/feeds/packages/net/noddos'
time: package/feeds/packages/noddos/compile#357.21#16.91#459.90
make[3]: Entering directory '/home/h/Desktop/lede/feeds/luci/applications/luci-app-noddos'
mkdir -p /home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/root-ar71xx/stamp
SHELL= flock /home/h/Desktop/lede/tmp/.root-copy.flock -c 'cp -fpR /home/h/Desktop/lede/build_dir/target-mips_24kc_musl/luci-app-noddos/.pkgdir/luci-app-noddos/. /home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/root-ar71xx/'
touch /home/h/Desktop/lede/staging_dir/target-mips_24kc_musl/root-ar71xx/stamp/.luci-app-noddos_installed
make[3]: Leaving directory '/home/h/Desktop/lede/feeds/luci/applications/luci-app-noddos'
time: package/feeds/luci/luci-app-noddos/compile#0.47#0.18#2.01
make[3]: Entering directory '/home/h/Desktop/lede/feeds/packages/net/ntpclient'

--- a/net/noddos/Makefile
+++ b/net/noddos/Makefile
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 # Name and release number of this package
 PKG_NAME:=noddos
-PKG_RELEASE:=2
+PKG_RELEASE:=3
 PKG_LICENSE:=GPLv3
 PKG_MAINTAINER:=Steven Hessing <steven.hessing@gmail.com>
 

--- a/net/noddos/patches/020-Ipset.cxx-update-libipset-API-to-version-7.patch
+++ b/net/noddos/patches/020-Ipset.cxx-update-libipset-API-to-version-7.patch
@@ -1,0 +1,431 @@
+From 5200105f412ceefa0784bf914aa215146fd067b0 Mon Sep 17 00:00:00 2001
+From: Eneas U de Queiroz <cote2004-github@yahoo.com>
+Date: Wed, 26 Dec 2018 16:45:47 -0200
+Subject: [PATCH] Ipset.cxx: update libipset API to version 7
+
+Old API compatibility was kept with a compatibility shim.
+
+Signed-off-by: Eneas U de Queiroz <cote2004-github@yahoo.com>
+
+diff --git a/src/Ipset.cxx b/src/Ipset.cxx
+index 9333fe6..da97f93 100644
+--- a/src/Ipset.cxx
++++ b/src/Ipset.cxx
+@@ -90,23 +90,19 @@ void Ipset::Open (const std::string inIpsetName, std::string inIpsetType, bool i
+     isIpsetv4 = inisIpsetv4;
+     ipset_load_types();
+ 
+-    struct ipset_session *session = ipset_session_init(printf);
++    struct ipset_session *session = noddos_ipset_session_init();
+     if (session == nullptr) {
+         syslog (LOG_ERR, "Ipset: Cannot initialize ipset session.");
+         ipset_session_fini(session);
+         throw std::runtime_error ("Cannot initialize ipset session.");
+     }
+ 
+-    if (ipset_envopt_parse(session, IPSET_ENV_EXIST, NULL) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't set environment option.");
+-        ipset_session_fini(session);
+-        throw std::runtime_error ("Can't set environment option.");
+-    }
++    ipset_envopt_set(session, IPSET_ENV_EXIST);
+     int r = ipset_session_data_set(session, IPSET_SETNAME, ipsetName.c_str());
+     if ( r < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't set setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't set setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't set setname " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't set setname " + ipsetName + ": " + ipset_session_report_msg(session));
+     } else if (r > 0) {
+         if (Debug == true) {
+             syslog (LOG_DEBUG, "Ipset: Not creating set %s as it already exists", ipsetName.c_str());
+@@ -115,27 +111,27 @@ void Ipset::Open (const std::string inIpsetName, std::string inIpsetType, bool i
+         return;
+     }
+     if (ipset_session_data_set(session, IPSET_OPT_TYPENAME, ipsetType.c_str()) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't set setname %s to type %s: %s", ipsetName.c_str(), ipsetType.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't set setname %s to type %s: %s", ipsetName.c_str(), ipsetType.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't set type " + ipsetType + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't set type " + ipsetType + ": " + ipset_session_report_msg(session));
+     }
+     const struct ipset_type *type = ipset_type_get(session, IPSET_CMD_CREATE);
+     if (type == NULL) {
+-        syslog (LOG_ERR, "Ipset: Can't set create ip %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't set create ip %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't create ipset " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't create ipset " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+ 
+     uint32_t timeout = 0; /* default to infinity */
+     if (ipset_session_data_set(session, IPSET_OPT_TIMEOUT, &timeout) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't set setname %s to timeout %d: %s", ipsetName.c_str(), timeout, ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't set setname %s to timeout %d: %s", ipsetName.c_str(), timeout, ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't set time-out " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't set time-out " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     if (ipset_session_data_set(session, IPSET_OPT_TYPE, type)) {
+-        syslog (LOG_ERR, "Ipset: Can't set setname %s option type: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't set setname %s option type: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't set ipset type: " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't set ipset type: " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     uint8_t family = 0;
+     if (ipsetType == "hash:ip" && isIpsetv4 == true) {
+@@ -149,20 +145,20 @@ void Ipset::Open (const std::string inIpsetName, std::string inIpsetType, bool i
+         throw std::invalid_argument("Unknown ipset data type " + ipsetType);
+     }
+     if (ipset_session_data_set(session, IPSET_OPT_FAMILY, &family) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't set setname %s address family %d: %s", ipsetName.c_str(), family, ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't set setname %s address family %d: %s", ipsetName.c_str(), family, ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Cannot set ipset family: " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Cannot set ipset family: " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+ 
+     if (ipset_cmd(session, IPSET_CMD_CREATE, /*lineno*/ 0) != 0) {
+-        syslog (LOG_ERR, "Ipset: Can't create setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't create setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Failed to create ipset " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Failed to create ipset " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     if (ipset_commit(session) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't commit for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't commit for setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't call ipset_commit for " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't call ipset_commit for " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     ipset_session_fini(session);
+ }
+@@ -173,33 +169,29 @@ bool Ipset::ipset_exec(enum ipset_cmd cmd) {
+     if (Debug == true) {
+         syslog(LOG_DEBUG, "Ipset: received command %d for ipset %s", cmd, ipsetName.c_str());
+     }
+-    struct ipset_session *session = ipset_session_init(printf);
++    struct ipset_session *session = noddos_ipset_session_init();
+     if (session == nullptr) {
+         syslog (LOG_ERR, "Ipset: Cannot initialize ipset session.");
+         ipset_session_fini(session);
+         throw std::runtime_error ("Cannot initialize ipset session.");
+     }
+ 
+-    if (ipset_envopt_parse(session, IPSET_ENV_EXIST, NULL) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't set environment option.");
+-        ipset_session_fini(session);
+-        throw std::runtime_error ("Can't set environment option.");
+-    }
++    ipset_envopt_set(session, IPSET_ENV_EXIST);
+     if (ipset_session_data_set(session, IPSET_SETNAME, ipsetName.c_str()) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't set setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't set setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't set setname " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't set setname " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+ 
+     if (ipset_cmd(session, cmd, 0) != 0) {
+         ipset_session_fini(session);
+-        syslog (LOG_ERR, "Ipset: Can't exec ipset cmd for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
+-        throw std::runtime_error("Can't exec ipset cmd for " + ipsetName + ": " + ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't exec ipset cmd for setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
++        throw std::runtime_error("Can't exec ipset cmd for " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     if (ipset_commit(session) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't commit for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't commit for setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't call ipset_commit for " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't call ipset_commit for " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     ipset_session_fini(session);
+     return true;
+@@ -210,61 +202,57 @@ bool Ipset::ipset_exec(enum ipset_cmd cmd,  const Tins::IPv4Address &inIpAddress
+     if (Debug == true) {
+         syslog(LOG_DEBUG, "Ipset: received command %d for IP address %s for ipset %s", cmd, inIpAddress.to_string().c_str(), ipsetName.c_str());
+     }
+-    struct ipset_session *session = ipset_session_init(printf);
++    struct ipset_session *session = noddos_ipset_session_init();
+     if (session == nullptr) {
+         syslog (LOG_ERR, "Ipset: Cannot initialize ipset session.");
+         ipset_session_fini(session);
+         throw std::runtime_error ("Cannot initialize ipset session.");
+     }
+ 
+-    if (ipset_envopt_parse(session, IPSET_ENV_EXIST, NULL) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't set environment option.");
+-        ipset_session_fini(session);
+-        throw std::runtime_error ("Can't set environment option.");
+-    }
++    ipset_envopt_set(session, IPSET_ENV_EXIST);
+     if (ipset_session_data_set(session, IPSET_SETNAME, ipsetName.c_str()) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't set setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't set setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't set setname " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't set setname " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     const struct ipset_type *type = ipset_type_get(session, cmd);
+     if (type == NULL) {
+-        syslog (LOG_ERR, "Ipset: Can't get type for set %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't get type for set %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't get type for set " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't get type for set " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+ 
+     uint8_t family = NFPROTO_IPV4;
+     if (ipset_session_data_set(session, IPSET_OPT_FAMILY, &family) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't set session data to IPv4 family for set %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't set session data to IPv4 family for set %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't set session data for " + ipsetName + " to the IPv4 family, error: " + ipset_session_error(session));
++        throw std::runtime_error("Can't set session data for " + ipsetName + " to the IPv4 family, error: " + ipset_session_report_msg(session));
+     }
+     struct in_addr sin;
+     inet_aton (inIpAddress.to_string().c_str(), &sin);
+     if (ipset_session_data_set(session, IPSET_OPT_IP, &sin) < 0) {
+-        syslog (LOG_ERR, "Can't set session data to the IPv4 address for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Can't set session data to the IPv4 address for setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't set session data to the IPv4 address for setname " + ipsetName + ", error: " + ipset_session_error(session));
++        throw std::runtime_error("Can't set session data to the IPv4 address for setname " + ipsetName + ", error: " + ipset_session_report_msg(session));
+     }
+ 
+     if (timeout) {
+         if (ipset_session_data_set(session, IPSET_OPT_TIMEOUT, &timeout) != 0) {
+-            syslog (LOG_ERR, "Ipset: Can't set timeout for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++            syslog (LOG_ERR, "Ipset: Can't set timeout for setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+             ipset_session_fini(session);
+-            throw std::runtime_error("Can't set timeout for " + ipsetName + ": " + ipset_session_error(session));
++            throw std::runtime_error("Can't set timeout for " + ipsetName + ": " + ipset_session_report_msg(session));
+             return false;
+         }
+     }
+     if (ipset_cmd(session, cmd, 0) != 0) {
+         ipset_session_fini(session);
+-        syslog (LOG_ERR, "Ipset: Can't exec ipset cmd for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
+-        throw std::runtime_error("Can't exec ipset cmd for " + ipsetName + ": " + ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't exec ipset cmd for setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
++        throw std::runtime_error("Can't exec ipset cmd for " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     if (ipset_commit(session) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't commit for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't commit for setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't call ipset_commit for " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't call ipset_commit for " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     ipset_session_fini(session);
+     return true;
+@@ -274,61 +262,57 @@ bool Ipset::ipset_exec(enum ipset_cmd cmd,  const Tins::IPv6Address &inIpAddress
+     if (Debug == true) {
+         syslog(LOG_DEBUG, "Ipset: received command %d for IP address %s for ipset %s", cmd, inIpAddress.to_string().c_str(), ipsetName.c_str());
+     }
+-    struct ipset_session *session = ipset_session_init(printf);
++    struct ipset_session *session = noddos_ipset_session_init();
+     if (session == nullptr) {
+         syslog (LOG_ERR, "Ipset: Cannot initialize ipset session.");
+         ipset_session_fini(session);
+         throw std::runtime_error ("Cannot initialize ipset session.");
+     }
+ 
+-    if (ipset_envopt_parse(session, IPSET_ENV_EXIST, NULL) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't set environment option.");
+-        ipset_session_fini(session);
+-        throw std::runtime_error ("Can't set environment option.");
+-    }
++    ipset_envopt_set(session, IPSET_ENV_EXIST);
+     if (ipset_session_data_set(session, IPSET_SETNAME, ipsetName.c_str()) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't set setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't set setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't set setname " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't set setname " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     const struct ipset_type *type = ipset_type_get(session, cmd);
+     if (type == NULL) {
+-        syslog (LOG_ERR, "Ipset: Can't get type for set %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't get type for set %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't get type for set " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't get type for set " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+ 
+     uint8_t family = NFPROTO_IPV6;
+     if (ipset_session_data_set(session, IPSET_OPT_FAMILY, &family) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't set session data to IPv6 family for set %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't set session data to IPv6 family for set %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't set session data for " + ipsetName + " to the IPv6 family, error: " + ipset_session_error(session));
++        throw std::runtime_error("Can't set session data for " + ipsetName + " to the IPv6 family, error: " + ipset_session_report_msg(session));
+     }
+ 
+     unsigned char buf[sizeof(struct in6_addr)];
+     int s = inet_pton(AF_INET6, inIpAddress.to_string().c_str(), buf);
+     if (ipset_session_data_set(session, IPSET_OPT_IP, &buf) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't set session data to the IPv4 address for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't set session data to the IPv4 address for setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't set session data to the IPv4 address for setname " + ipsetName + ", error: " + ipset_session_error(session));
++        throw std::runtime_error("Can't set session data to the IPv4 address for setname " + ipsetName + ", error: " + ipset_session_report_msg(session));
+     }
+ 
+     if (timeout) {
+         if (ipset_session_data_set(session, IPSET_OPT_TIMEOUT, &timeout) != 0) {
+-            syslog (LOG_ERR, "Ipset: Can't set timeout for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++            syslog (LOG_ERR, "Ipset: Can't set timeout for setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+             ipset_session_fini(session);
+-            throw std::runtime_error("Can't set timeout for " + ipsetName + ": " + ipset_session_error(session));
++            throw std::runtime_error("Can't set timeout for " + ipsetName + ": " + ipset_session_report_msg(session));
+         }
+     }
+     if (ipset_cmd(session, cmd, 0) != 0) {
+-        syslog (LOG_ERR, "Ipset: Can't exec ipset cmd for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't exec ipset cmd for setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't exec ipset cmd for " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't exec ipset cmd for " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     if (ipset_commit(session) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't commit for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't commit for setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't call ipset_commit for " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't call ipset_commit for " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     ipset_session_fini(session);
+     return true;
+@@ -338,50 +322,46 @@ bool Ipset::ipset_exec(enum ipset_cmd cmd, const std::string Mac, time_t timeout
+     if (Debug == true) {
+         syslog(LOG_DEBUG, "Ipset: received command %d for MAC address %s for ipset %s", cmd, Mac.c_str(), ipsetName.c_str());
+     }
+-    struct ipset_session *session = ipset_session_init(printf);
++    struct ipset_session *session = noddos_ipset_session_init();
+     if (session == nullptr) {
+         syslog (LOG_ERR, "Ipset: Cannot initialize ipset session.");
+         ipset_session_fini(session);
+         throw std::runtime_error ("Cannot initialize ipset session.");
+     }
+ 
+-    if (ipset_envopt_parse(session, IPSET_ENV_EXIST, NULL) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't set environment option.");
+-        ipset_session_fini(session);
+-        throw std::runtime_error ("Can't set environment option.");
+-    }
++    ipset_envopt_set(session, IPSET_ENV_EXIST);
+     if (ipset_session_data_set(session, IPSET_SETNAME, ipsetName.c_str()) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't set setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't set setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't set setname " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't set setname " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     const struct ipset_type *type = ipset_type_get(session, cmd);
+     if (type == NULL) {
+-        syslog (LOG_ERR, "Ipset: Can't get type for set %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't get type for set %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't get type for set " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't get type for set " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     if (ipset_parse_elem(session, (ipset_opt)type->last_elem_optional, Mac.c_str()) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't call ipset_parse_elem for %s: %s ", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't call ipset_parse_elem for %s: %s ", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't call ipset_parse_elem for ipset " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't call ipset_parse_elem for ipset " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     if (timeout) {
+         if (ipset_session_data_set(session, IPSET_OPT_TIMEOUT, &timeout) != 0) {
+-            syslog (LOG_ERR, "Ipset: Can't set timeout for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++            syslog (LOG_ERR, "Ipset: Can't set timeout for setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+             ipset_session_fini(session);
+-            throw std::runtime_error("Can't set timeout for " + ipsetName + ": " + ipset_session_error(session));
++            throw std::runtime_error("Can't set timeout for " + ipsetName + ": " + ipset_session_report_msg(session));
+         }
+     }
+     if (ipset_cmd(session, cmd, 0) != 0) {
+-        syslog (LOG_ERR, "Ipset: Can't exec ipset cmd for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't exec ipset cmd for setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't exec ipset cmd for " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't exec ipset cmd for " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     if (ipset_commit(session) < 0) {
+-        syslog (LOG_ERR, "Ipset: Can't commit for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++        syslog (LOG_ERR, "Ipset: Can't commit for setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+         ipset_session_fini(session);
+-        throw std::runtime_error("Can't call ipset_commit for " + ipsetName + ": " + ipset_session_error(session));
++        throw std::runtime_error("Can't call ipset_commit for " + ipsetName + ": " + ipset_session_report_msg(session));
+     }
+     ipset_session_fini(session);
+     return true;
+diff --git a/src/Ipset.h b/src/Ipset.h
+index 2c5f7b2..eb180da 100644
+--- a/src/Ipset.h
++++ b/src/Ipset.h
+@@ -41,6 +41,31 @@
+ 
+ #include "MacAddress.h"
+ 
++#if IPSET_PROTOCOL < 7
++/* compatibility shims */
++
++inline void ipset_envopt_set(struct ipset_session *session, enum ipset_envopt opt)
++{
++    ipset_envopt_parse(session, opt, NULL);
++}
++
++inline const char * ipset_session_report_msg(const struct ipset_session *session)
++{
++  return ipset_session_error(session);
++}
++
++static inline struct ipset_session *noddos_ipset_session_init(void)
++{
++  return ipset_session_init(printf);
++}
++
++#else
++
++static inline struct ipset_session *noddos_ipset_session_init(void)
++{
++  return ipset_session_init(NULL, NULL);
++}
++#endif
+ 
+ std::string getIpsetUuid (std::string inUuid);
+ std::string getIpsetName (std::string inUuid, bool inSrc, bool inIpv4 = true);
+@@ -99,23 +124,19 @@ public:
+     }
+     bool Exists() {
+         try {
+-            struct ipset_session *session = ipset_session_init(printf);
++            struct ipset_session *session = noddos_ipset_session_init();
+             if (session == nullptr) {
+                 syslog (LOG_ERR, "Ipset: Cannot initialize ipset session.");
+                 ipset_session_fini(session);
+                 throw std::runtime_error ("Cannot initialize ipset session.");
+             }
+ 
+-            if (ipset_envopt_parse(session, IPSET_ENV_EXIST, NULL) < 0) {
+-                syslog (LOG_ERR, "Ipset: Can't set environment option.");
+-                ipset_session_fini(session);
+-                throw std::runtime_error ("Can't set environment option.");
+-            }
++            ipset_envopt_set(session, IPSET_ENV_EXIST);
+             int r = ipset_session_data_set(session, IPSET_SETNAME, ipsetName.c_str());
+             if (ipset_commit(session) < 0) {
+-                syslog (LOG_ERR, "Ipset: Can't commit for setname %s: %s", ipsetName.c_str(), ipset_session_error(session));
++                syslog (LOG_ERR, "Ipset: Can't commit for setname %s: %s", ipsetName.c_str(), ipset_session_report_msg(session));
+                 ipset_session_fini(session);
+-                throw std::runtime_error("Can't call ipset_commit for " + ipsetName + ": " + ipset_session_error(session));
++                throw std::runtime_error("Can't call ipset_commit for " + ipsetName + ": " + ipset_session_report_msg(session));
+             }
+             ipset_session_fini(session);
+             return r == 0;

--- a/net/noddos/patches/030-getnoddosdeviceprofiles-wget-timestamping-check.patch
+++ b/net/noddos/patches/030-getnoddosdeviceprofiles-wget-timestamping-check.patch
@@ -1,0 +1,28 @@
+From eb1730afff9377a5f167d0738ad0b3aeba9634d0 Mon Sep 17 00:00:00 2001
+From: Eneas U de Queiroz <cote2004-github@yahoo.com>
+Date: Tue, 19 Mar 2019 18:27:10 -0300
+Subject: [PATCH] getnoddosdeviceprofiles: wget timestamping check
+
+Check if the --timestamping option is available to avoid an error in
+openwrt when wget is handled by uclient-fetch.
+
+Signed-off-by: Eneas U de Queiroz <cote2004-github@yahoo.com>
+
+diff --git a/tools/getnoddosdeviceprofiles b/tools/getnoddosdeviceprofiles
+index 337e351..174034f 100755
+--- a/tools/getnoddosdeviceprofiles
++++ b/tools/getnoddosdeviceprofiles
+@@ -86,7 +86,12 @@ fi
+ # That's also why we don't delete the downloaded file
+ if [ "$WGET" != "" ]
+ then
+-    GETURL="$WGET --quiet --timestamping"
++    GETURL="$WGET --quiet"
++    # Make sure wget accepts --timestamping
++    if wget --help 2>&1 | egrep timestamping > /dev/null
++    then
++        GETURL="$GETURL --timestamping"
++    fi
+ else
+     if [ "$CURL" != "" ]
+     then

